service/fit.py

```diff
@@ -1125,8 +1128,41 @@ def refreshFit(self, fitID):
         self.recalc(fit)
 
     def recalc(self, fit, withBoosters=True):
+        global recalcJob
+
+        if not schedule.empty():
+            if recalcJob in schedule.queue:
+                logger.debug("Job already scheduled. Cancelling existing job")
+                schedule.cancel(recalcJob)
+                pass
```

Comment:
I know I don't need the 
```
     pass
else:
     pass
```
here.  This is just simply for some flow control, and I was experimenting with different ways of handling creating the job.